google/cloud/storage/internal/tracing_client.cc

```diff
@@ -0,0 +1,430 @@
+  auto span = internal::MakeSpan("storage::Client::PatchBucket");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->PatchBucket(request));
+}
+
+StatusOr<storage::NativeIamPolicy> TracingClient::GetNativeBucketIamPolicy(
+    storage::internal::GetBucketIamPolicyRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::GetNativeBucketIamPolicy");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->GetNativeBucketIamPolicy(request));
+}
+
+StatusOr<storage::NativeIamPolicy> TracingClient::SetNativeBucketIamPolicy(
+    storage::internal::SetNativeBucketIamPolicyRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::SetNativeBucketIamPolicy");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->SetNativeBucketIamPolicy(request));
+}
+
+StatusOr<storage::internal::TestBucketIamPermissionsResponse>
+TracingClient::TestBucketIamPermissions(
+    storage::internal::TestBucketIamPermissionsRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::TestBucketIamPermissions");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->TestBucketIamPermissions(request));
+}
+
+StatusOr<storage::BucketMetadata> TracingClient::LockBucketRetentionPolicy(
+    storage::internal::LockBucketRetentionPolicyRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::LockBucketRetentionPolicy");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->LockBucketRetentionPolicy(request));
+}
+
+StatusOr<storage::ObjectMetadata> TracingClient::InsertObjectMedia(
+    storage::internal::InsertObjectMediaRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::InsertObjectMedia");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->InsertObjectMedia(request));
+}
+
+StatusOr<storage::ObjectMetadata> TracingClient::CopyObject(
+    storage::internal::CopyObjectRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::CopyObject");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->CopyObject(request));
+}
+
+StatusOr<storage::ObjectMetadata> TracingClient::GetObjectMetadata(
+    storage::internal::GetObjectMetadataRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::GetObjectMetadata");
+  auto scope = opentelemetry::trace::Scope(span);
+  return internal::EndSpan(*span, impl_->GetObjectMetadata(request));
+}
+
+StatusOr<std::unique_ptr<storage::internal::ObjectReadSource>>
+TracingClient::ReadObject(
+    storage::internal::ReadObjectRangeRequest const& request) {
+  auto span = internal::MakeSpan("storage::Client::ReadObject");
+  auto scope = opentelemetry::trace::Scope(span);
+  // TODO(#.....) - add a wrapper for ReadObjectSource.
```

Comment:
I am surprised `clang-tidy` let you get away with this.

Reply:
Fixed. `clang-tidy` and `checkers` do not know anything about TODO entries.